src/test/java/com/spotify/dns/DnsSrvWatchersTest.java

```diff
@@ -0,0 +1,90 @@
+package com.spotify.dns;
+
+import com.google.common.collect.ImmutableList;
+
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.not;
+
+public class DnsSrvWatchersTest {
+
+  @Test
+  public void triggerRace() throws Exception {
+    int limit = 100000;
+    while (limit-- > 0) {
+      System.out.println(limit);
+      resolve();
+    }
+  }
+
+  private void resolve() throws Exception {
```

Comment:
I think it would be nice to keep this test, if it can be made to fail instead of hang. I guess await with a timeout should do it? Or actually, if the problem is that updates arrive out of order (rather than don't arrive at all), then why does it hang?